src/utils/cspr-network.ts

```diff
@@ -0,0 +1,240 @@
+            delegator: CLValue.newCLPublicKey(delegatorPublicKey),
+            amount: CLValueUInt512.newCLUInt512(amountMotes)
+          })
+        )
+        .buildFor1_5();
+    }
+
+    throw new Error(
+      'Auction contract hash is required when creating a transaction on Casper Network 1.5.x'
+    );
+  }
+
+  public createRedelegateTransaction(
+    delegatorPublicKey: PublicKey,
+    validatorPublicKey: PublicKey,
+    newValidatorPublicKey: PublicKey,
+    networkName: string,
+    amountMotes: string | BigNumber,
+    deployCost: number,
+    ttl: number,
+    auctionContractHash?: string
+  ): Transaction {
+    if (this.apiVersion === 2) {
+      new NativeRedelegateBuilder()
+        .validator(validatorPublicKey)
+        .newValidator(newValidatorPublicKey)
+        .from(delegatorPublicKey)
+        .amount(amountMotes)
+        .chainName(networkName)
+        .payment(deployCost)
+        .ttl(ttl)
+        .build();
+    }
+
+    if (auctionContractHash) {
+      return new ContractCallBuilder()
+        .from(delegatorPublicKey)
+        .byHash(auctionContractHash)
+        .entryPoint('redelegate')
+        .chainName(networkName)
+        .runtimeArgs(
+          Args.fromMap({
+            validator: CLValue.newCLPublicKey(validatorPublicKey),
+            delegator: CLValue.newCLPublicKey(delegatorPublicKey),
+            amount: CLValueUInt512.newCLUInt512(amountMotes),
+            ...(newValidatorPublicKey
+              ? {
+                  new_validator: CLValue.newCLPublicKey(newValidatorPublicKey)
+                }
+              : {})
+          })
+        )
+        .ttl(ttl)
+        .buildFor1_5();
+    }
+
+    throw new Error(
+      'Auction contract hash is required when creating a transaction on Casper Network 1.5.x'
+    );
+  }
+
+  public createTransferTransaction(
+    senderPublicKey: PublicKey,
+    recepientPublicKey: PublicKey,
+    networkName: string,
+    amountMotes: string,
+    deployCost: number,
+    ttl: number
+  ): Transaction {
+    const transferBuilder = new NativeTransferBuilder()
+      .from(senderPublicKey)
+      .target(recepientPublicKey)
+      .amount(amountMotes)
+      .chainName(networkName)
+      .payment(deployCost)
+      .ttl(ttl);
+    if (this.apiVersion === 2) {
+      return transferBuilder.build();
+    }
+
+    return transferBuilder.buildFor1_5();
+  }
+
+  public async putTransaction(
+    transaction: Transaction
+  ): Promise<PutTransactionResult | PutDeployResult> {
+    if (this.apiVersion == 2) {
+      return await this.rpcClient.putTransaction(transaction);
+    }
+
+    const deploy = transaction.getDeploy();
+    if (deploy) {
+      return await this.rpcClient.putDeploy(deploy);
+    }
+
+    return Promise.reject(
+      'Legacy deploy transaction is required when submitting to Casper Network 1.5'
+    );
+  }
+
+  public async getTransaction(
+    hash: TransactionHash
+  ): Promise<InfoGetTransactionResult | InfoGetDeployResult> {
```

Comment:
We need to unify this return type to `InfoGetTransactionResult`. It should be possible to do since we can convert `Deploy` to `Transaction` and `ExecutionResults` to `ExecutionResultsV2`